src/main/java/pl/umk/mat/git2befit/service/workout/factory/implementation/FBWTrainingPlan.java

```diff
@@ -85,18 +90,20 @@ public void validateAfterCreating() {
             Collections.shuffle(exercisesWithEquipmentFilteredByBodyPart);
 
             for (int i = 0; i < localTrainingForm.getDaysCount(); i++) {
-                Exercise exercise;
-                if (isEnough(exercisesWithEquipmentFilteredByBodyPart))
-                    exercise = exercisesWithEquipmentFilteredByBodyPart.remove(i);
-                else
-                    exercise = exercisesWithEquipmentFilteredByBodyPart.get(
-                            i % exercisesWithEquipmentFilteredByBodyPart.size()
-                    );
-
-                exerciseExecutionList.add(getExactExerciseExecution(
-                        exercise,
-                        this.trainingForm
-                ));
+                if(exercisesWithEquipmentFilteredByBodyPart.size() != 0) {
+                    Exercise exercise;
+                    if (isEnough(exercisesWithEquipmentFilteredByBodyPart))
+                        exercise = exercisesWithEquipmentFilteredByBodyPart.remove(i);
+                    else
+                        exercise = exercisesWithEquipmentFilteredByBodyPart.get(
+                                i % exercisesWithEquipmentFilteredByBodyPart.size()
+                        );
```

Comment:
Dlaczego w elsie nie usuwamy ćwiczenia?

Reply:
bo działa to na takiej zasdzie, że jak będzie mniej ćwiczeń(np 2) niż dni(np 3) to będzie brał z puli tych 2 ćwiczeń, i nie dojedzie do sytuacji, że jakiś dzień nie ma ćwiczneia

Reply:
Przedyskutowałbym to jeszcze na spotkaniu. Implementacja zrozumiała, ale czy zgadza się to B I Z N E S O W O